### DISCUSSION TITLE
Missing libstt.so file for Android ?

### DISCUSSION THREAD

**Discussion by GrzesiowVRze** [coqui-ai/STT#2126] [Q&A] [answered]
Hi,
I am trying to implement Speech Recognition system in Unity in VR application, that will be run on mobile googles like Oculus Quest and Pico Neo, which runs on Android.
I am following this tutorial right here, for implementing cogui in Unity:
https://bitbarrelmedia.wordpress.com/2021/09/19/coqui-stt-in-unity/
This tutorial mention about libstt.so file for every platform, like libstt.so in native_client.tflite.Windows.tar.xz for Windows.
I can't find libstt.so file or native_client zip or folder for Android.
Am I just blind, or maybe I need to somehow build it myself? If so, can you link me some tutorial or documentation how to do it?
Thanks in advance,
Grzesio

**Comment by reuben** [comment, marked as answer]
For Android we release an `.aar` package which has the library embedded in it. For Coqui STT 1.2.0 it's called libstt-1.2.0-release.aar in the [release assets](https://github.com/coqui-ai/STT/releases/v1.2.0).

**Reply by GrzesiowVRze** [reply]
Thanks, I found it :)

**Comment by GrzesiowVRze** [comment]
Hi, 
Thanks for reply, I found it, but now I'm dealing with an error: DllNotFoundException: libstt.so. See attached image for details
![coqui stt issue](https://user-images.githubusercontent.com/100303490/155749147-dc100818-d92a-4c42-9603-62b378e8d82e.PNG)
.
I took libstt.so file from libstt-1.2.0-release.aar package how you told me, and put it in CoquiSTT/Plugins/android folder in Unity project, so it should run on builds for Android. But maybe I need to copy it to some other folder to make it run. Do you have some thoughts about this? I know that you probably don't dealing with Coqui in Unity, but maybe you can give some advice about this error.
Thanks in advance,
Grzesio

**Reply by reuben** [reply]
I don't know about Unity integration, but the AAR is meant to be integrated using Gradle, you shouldn't need to extract it and grab binaries by hand. The native library inside of it is picked up automatically.
